### PR TITLE
10.2.0 capabilities

### PR DESCRIPTION
**Pull request by Expressionless**
Implemented Search Capabilities. Tested serialization and implemented traitCapabilitiesMet function. Had issues with SearchEntity serialisation following converting to a Map for storage of traits, but all works now. I can tune SearchEntity further as required if necessary

### CODE REVIEW

File: qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java [hunk 1]
```diff
@@ -15,46 +17,52 @@
...
+@SuppressWarnings("unchecked")
 @RegisterForReflection
 public class SearchEntity extends BaseEntity {
 
 	private static final Logger log = Logger.getLogger(SearchEntity.class);
 
 	private static final long serialVersionUID = 1L;
 
+	// TODO: Polish this
+	private Map<Class<? extends Trait>, List<? extends Trait>> traits = new HashMap<>();
```
Comment (jasperRob) [inline qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java:43]: Does this map retain the indexed order that sorts and columns, etc. were added in?

File: qwandaq/src/main/docker/Dockerfile.jvm [hunk 2]
```diff
@@ -75,9 +75,10 @@
 #   accessed directly. (example: "foo.example.com,bar.example.com")
 #
 ###
-FROM registry.access.redhat.com/ubi8/openjdk-11:1.11
+FROM registry.access.redhat.com/ubi8/openjdk-17:1.11
```
Comment (jasperRob) [inline qwandaq/src/main/docker/Dockerfile.jvm]: Are we moving to Java17?

File: qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java [hunk 3]
```diff
@@ -15,46 +17,52 @@
...
+
 	private List<ClauseContainer> clauseContainers = new ArrayList<>();
-	private List<Sort> sorts = new ArrayList<>();
-	private List<Column> columns = new ArrayList<>();
-	private List<Action> actions = new ArrayList<>();
+	// private List<Sort> sorts = new ArrayList<>();
+	// private List<Column> columns = new ArrayList<>();
+	// private List<Action> actions = new ArrayList<>();
```
Comment (jasperRob) [inline qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java]: Might as well delete these if not used anymore

**Review by jasperRob — CHANGES_REQUESTED** [review]
Couple of changes

File: qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java (hunk 1 above)
Comment (Expressionless) [reply qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java:43]: It does, given it stores a list as a value the order will be retained

File: qwandaq/src/main/docker/Dockerfile.jvm (hunk 2 above)
Comment (Expressionless) [reply qwandaq/src/main/docker/Dockerfile.jvm]: Ahh this may have bled through from somewhere else. Not to my knowledge

**Review by jasperRob — APPROVED** [review]
LGTM

**Review by varunshastry — APPROVED** [review]

**Pull request merged**